packages/block-library/src/template-part/edit/index.js

```diff
@@ -67,11 +69,52 @@ function ReplaceButton( {
 	);
 }
 
+function NonEditableTemplatePartPreview( {
+	postId: id,
+	layout,
+	tagName: TagName,
+	blockProps,
+} ) {
+	const themeSupportsLayout = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return getSettings()?.supportsLayout;
+	}, [] );
+	const [ defaultLayout ] = useSettings( 'layout' );
+	const usedLayout = layout?.inherit ? defaultLayout || {} : layout;
+
+	const [ blocks ] = useEntityBlockEditor( 'postType', 'wp_template_part', {
```

Comment:
Can't we just retrieve the "blocks" property from the entity without using this hook.